### PR TITLE
Kamu UI 376 handle uncacheable sources in web UI

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**
Closes:https://github.com/kamu-data/kamu-web-ui/issues/376

Result:

image

image

### CODE REVIEW

File: src/app/common/components/flows-table/flows-table.component.ts [hunk 1]
```diff
@@ -76,10 +82,16 @@ export class FlowsTableComponent implements OnInit, OnChanges {
     public dropdownStatustList: FilterStatusType[] = [];
     public selectedStatusItems: FilterStatusType[] = [];
 
+    private scheduleOptions: IngestConditionInput;
```
Comment (zaychenko-sergei) [inline src/app/common/components/flows-table/flows-table.component.ts:85]: You don't need this variable, it's used only temporarily, so can become local.

File: src/app/common/components/flows-table/flows-table.component.ts [hunk 2]
```diff
@@ -196,6 +208,57 @@ export class FlowsTableComponent implements OnInit, OnChanges {
...
+            this.datasetFlowsService
+                .datasetTriggerFlow({
+                    datasetId: node.description.datasetId,
+                    datasetFlowType: DatasetFlowType.Ingest,
+                    flowRunConfiguration: {
+                        ingest: this.scheduleOptions,
+                    },
+                })
+                .pipe(takeUntilDestroyed(this.destroyRef))
+                .subscribe((result: boolean) => {
```
Comment (zaychenko-sergei) [inline src/app/common/components/flows-table/flows-table.component.ts:226]: Error handling?

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html [hunk 3]
```diff
@@ -18,6 +17,10 @@ <h4>Root polling source</h4>
                 aria-label="Select an option"
                 color="primary"
             >
+                <div class="d-flex align-items-center mt-4">
+                    <input class="checkbox-uncacheable me-3" type="checkbox" formControlName="fetchUncacheable" />
+                    <span class="d-inline-block">Fetch uncachable</span>
+                </div>
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html:23]: It's a little strange this became a top option. It's quite a rare detail, so could be below the main set of options.

File: src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts [hunk 4]
```diff
@@ -136,10 +137,17 @@ export class DatasetFlowDetailsHelpers {
                                                       ? "block"
                                                       : "blocks"
                                               }`
-                                            : "Dataset is up-to-date";
+                                            : flowDetails.description.ingestResult?.__typename ===
+                                                    "FlowDescriptionUpdateResultUpToDate" &&
+                                                flowDetails.description.ingestResult.uncacheable &&
+                                                flowDetails.configSnapshot?.__typename === "FlowConfigurationIngest" &&
+                                                !flowDetails.configSnapshot.fetchUncacheable
+                                              ? "Source is uncacheable, use force update to re-scan the data"
```
Comment (zaychenko-sergei) [inline src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts:145]: Why the template here different than above?

Comment (dmitriy-borzenko) [reply src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.helpers.ts:145]: Fixed to message as for the main table

File: src/app/common/components/flows-table/flows-table.component.html [hunk 5]
```diff
@@ -74,12 +74,27 @@
...
+                        ></div>
+                        &nbsp;
+                        <a
+                            *ngIf="
+                                element.description.ingestResult?.__typename ===
+                                    'FlowDescriptionUpdateResultUpToDate' &&
+                                element.description.ingestResult.uncacheable &&
+                                element.configSnapshot?.__typename === 'FlowConfigurationIngest' &&
+                                !element.configSnapshot.fetchUncacheable
+                            "
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/common/components/flows-table/flows-table.component.html:92]: Moved logic to component

File: src/app/common/components/flows-table/flows-table.component.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/common/components/flows-table/flows-table.component.ts:226]: Error handling exists in the service method

image

File: src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/dataset-settings-component/tabs/scheduling/dataset-settings-scheduling-tab.component.html]: Fixed

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**